### PR TITLE
F660 - custom loader

### PR DESCRIPTION
**Pull request by creiser2**

### CODE REVIEW

File: CranberryQueue/Shared/activityIndicatorPresenter.swift [hunk 1]
```diff
@@ -11,33 +11,81 @@ import Foundation
...
+        } else {
+           activityIndicator.imageView?.image = activityIndicator.ballForever
         }
     }
 
     func hideActivityIndicator() {
         DispatchQueue.main.async {
-            self.activityIndicator.stopAnimating()
-            self.activityIndicator.removeFromSuperview()
+            if let activityIndicator = self.view.viewWithTag(100) {
```
Comment (rolflocher) [inline CranberryQueue/Shared/activityIndicatorPresenter.swift]: oh yikes but I see why you did that.

consider another optional approach without starting this tag pattern.

if let activityIndicator = self.view.subviews.first(where: { $0 is CranberryActivityIndicator }) {

}

File: CranberryQueue/Shared/activityIndicatorPresenter.swift [hunk 2]
```diff
@@ -11,33 +11,81 @@ import Foundation
...
+        super.init(frame: frame)
+        let imgView = UIImageView()
+        imgView.frame = CGRect(x: bounds.size.width - (bounds.size.width/2), y: bounds.size.height - (bounds.size.height/2), width: 80, height: 80)
+        addSubview(imgView)
+        imageView = imgView
+        initAnimations()
+    }
+    
+    required init(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
```
Comment (rolflocher) [inline CranberryQueue/Shared/activityIndicatorPresenter.swift]: nice

File: CranberryQueue/Shared/activityIndicatorPresenter.swift [hunk 3]
```diff
@@ -11,33 +11,81 @@ import Foundation
...
+public class CranberryActivityIndicator: UIView {
+    var blackBallForeverSequence: [UIImage] = []
+    var ballForeverSequence: [UIImage] = []
+    var blackBallForever: UIImage? = UIImage(named: "Comp 1_00024")
+    var ballForever: UIImage? = nil
+    var imageView: UIImageView? = nil
+    
+    var cranberryActivityIndicator:CranberryActivityIndicator {
+        get {
+          return self
```
[the remark this answers is not in the document]
Comment (creiser2) [reply CranberryQueue/Shared/activityIndicatorPresenter.swift]: Lol how does this work haha

**Comment by Matt-Innocenzo** [comment]
We're thinking no white fill and should show on open

**Comment by creiser2** [comment]
@Matt-Innocenzo tried the no white fill originally. Unfortunately, depending on the map thats behind the loader, it can be much harder to see. That's why I switched to white fill.

**Pull request merged**